src/sidebar/test/media-embedder-test.js

```diff
@@ -30,6 +30,106 @@ describe('media-embedder', function () {
     });
   });
 
+  it('allows whitelisted parameters in YouTube watch URLs', function () {
+    var urls = [
+      'https://www.youtube.com/watch?v=QCkm0lL-6lc&start=5&end=10',
+      'https://www.youtube.com/watch?v=QCkm0lL-6lc&end=10&start=5',
+      'https://www.youtube.com/watch/?v=QCkm0lL-6lc&end=10&start=5',
+    ];
+    urls.forEach(function (url) {
+      var element = domElement('<a href="' + url + '">' + url + '</a>');
+
+      mediaEmbedder.replaceLinksWithEmbeds(element);
+
+      assert.equal(element.childElementCount, 1);
+      assert.equal(element.children[0].tagName, 'IFRAME', url);
+      // queryString's #stringify sorts keys, so resulting query string
+      // will be reliably as follows, regardless of original ordering
+      // Note also `v` param is handled elsewhere and is not "allowed" in
+      // queryString.
+      assert.equal(
+        element.children[0].src,
+        'https://www.youtube.com/embed/QCkm0lL-6lc?end=10&start=5');
+    });
+  });
+
+  it('translates YouTube watch `t` param to `start` for embed', function () {
+    var urls = [
+      'https://www.youtube.com/watch?v=QCkm0lL-6lc&t=5&end=10',
+      'https://www.youtube.com/watch?v=QCkm0lL-6lc&end=10&t=5',
+      'https://www.youtube.com/watch/?v=QCkm0lL-6lc&end=10&t=5',
+    ];
+    urls.forEach(function (url) {
+      var element = domElement('<a href="' + url + '">' + url + '</a>');
+
+      mediaEmbedder.replaceLinksWithEmbeds(element);
+
+      assert.equal(element.childElementCount, 1);
+      assert.equal(element.children[0].tagName, 'IFRAME', url);
+      assert.equal(
+        element.children[0].src,
+        'https://www.youtube.com/embed/QCkm0lL-6lc?end=10&start=5');
+    });
+  });
+
+  it('translates YouTube `t` param values to seconds', function () {
+    var urls = [
```

Comment:
Perhaps fold these two arrays into one so that the test case expectations and results can be more easily matched together. The entries could be tuples or objects.

```
var cases = [
  [url, embed],
  [url2, embed],
];
```